### PR TITLE
3. As a user, I want to set up a new shopping list so I can track purchased items (continued from week 2)

### PR DESCRIPTION
**Pull request by ajiles91**
just creating initial pull request

### CODE REVIEW

File: src/pages/List.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ const List = () => {
         path="items"
         // Sort the data
         sort="name"
+        // filter={['token', '==', 'baron flew gibby']}
         // isLoading = is a Boolean that represents the loading status for the firebase query. true until an initial payload from Firestore is received.
```
Comment (prophen) [inline src/pages/List.js]: @ajiles91  I know what the problem is. This is creating a compound query (order by name, then filter by token) and an index needs to be created for it to work.

I'm on it now. Nothing was wrong with our syntax, it's a Firebase thing.

Comment (prophen) [reply src/pages/List.js]: Also stackoverflow to the rescue - https://stackoverflow.com/questions/58261612/firestore-orderby-and-where-filters-not-working-together

Comment (prophen) [reply src/pages/List.js]: @segdeha @ajiles91 We manually added the token to these items in the database and displayed them using the hard-coded token for the query. Does this satisfy the story? 
List associated with token

Comment (segdeha) [reply src/pages/List.js]: This is on the right track! What would be better in terms of completing this in a way that let's us continue to build out the app is to read the token from `localStorage` and filter using that. So, to test this, you could manually set the token in `localStorage` and make sure your list pulls the right items. Let me know if you have any questions about that!

image

Comment (prophen) [reply src/pages/List.js]: That makes perfect sense to me. Thanks Andrew. cc @ajiles91

File: src/pages/List.js [hunk 2]
```diff
@@ -5,21 +5,8 @@ import Loading from '../components/Loading';
...
-  //   .get()
-  //   .then(function(querySnapshot) {
-  //     querySnapshot.forEach(function(doc) {
-  //       // doc.data() is never undefined for query doc snapshots
-  //       console.log(doc.id, ' => ', doc.data());
-  //     });
-  //   })
-  //   .catch(function(error) {
-  //     console.log('Error getting documents: ', error);
-  //   });
```
Comment (prophen) [inline src/pages/List.js]: I removed the code I added to test the query.

File: src/pages/List.js [hunk 3]
```diff
@@ -31,7 +18,7 @@ const List = () => {
         path="items"
         // Sort the data
         sort="name"
-        filter={['token', '==', 'baron flew gibby']}
+        filter={['token', '==', getStoredToken()]}
```
Comment (prophen) [inline src/pages/List.js]: No more hard coded token. It's using the token stored in local storage. I tested this by starting another server on a new port and adding new items to the shopping list.

List associated with token

File: src/components/NavTabs.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import '../pages/HomePage.css';
+
+const NavTabs = () => {
```
Comment (prophen) [inline src/components/NavTabs.js]: @ajiles91 @MonicaDJohnson @mikeramz86 I put the navigation tabs in a new component and included it on every page we have so far. Do you approve of this change? I didn't change any code.

Here's a sandbox showing the changes.
https://codesandbox.io/s/github/the-collab-lab/tcl-3-smart-shopping-list/tree/aj-np-create-new-list-token/?fontsize=14&hidenavigation=1&theme=dark

**Comment by prophen** [comment]
Finishing up the story from last week - https://github.com/the-collab-lab/tcl-3-smart-shopping-list/wiki/(Week-2)-Monica-&-Nikema-(As-a-user,-I-want-to-set-up-a-new-shopping-list)

PR: https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/49
****
A shopping list consists of a set of items associated with a user’s token. Tokens can be shared with other users to allow them to co-manage a given list. Creating a new list consists of the following:

- Generate a new unique token
- Save the token to localStorage
- Show the user the list view

The following script can be used to generate a suitable token: https://gist.github.com/segdeha/21a42618ce5a54916c5b58d36ec2992e
****
The requested changes from Andrew:

This is a great start, but is a little incomplete. Importantly, from the story description:

 A shopping list consists of a set of items associated with a user’s token.

What I'd like to see added to what you have is the following:

- Once the user has a token, redirect them back to / (our list view)
- On the list view, show only items associated with the user's token (should be able to add this as a parameter to the Firestore query)

You may need to add some records to the database directly to get them to display. In the next set of stories, we'll address adding items to the list in a way that stores the user's token as well (I'm a bit confused how that's already implemented because that story hasn't come up yet)

File: src/pages/AddItem.js [hunk 5]
```diff
@@ -1,13 +1,27 @@
...
   const [name, setName] = useState('');
 
+  const initialToken = () => window.localStorage.getItem('token') || getToken();
+  const [token] = useState(initialToken);
+  // const[ redirect, setRedirect ] = useState(false)
+  useEffect(() => {
+    window.localStorage.setItem('token', token);
+  }, [token]);
+
+  console.log('token:', token);
```
Comment (segdeha) [inline src/pages/AddItem.js]: Before merging, please remove `console.log` statements and commented out code. Thanks!

File: src/pages/HomePage.js [hunk 6]
```diff
@@ -1,22 +1,12 @@
...
-          <NavLink className="left-tab" to="/add-item">
-            to add item
-          </NavLink>
-
-          <NavLink className="right-tab" to="/list">
-            to list
-          </NavLink>
-        </div>
-      </div>
+      <NavTabs />
```
Comment (segdeha) [inline src/pages/HomePage.js]: Big improvement moving this to a reusable component!

File: src/pages/List.js [hunk 7]
```diff
@@ -1,19 +1,20 @@
 import React from 'react';
 import { FirestoreCollection } from 'react-firestore';
-import { Link } from 'react-router-dom';
 import Loading from '../components/Loading';
+import NavTabs from '../components/NavTabs';
+// import firebase from 'firebase/app';
```
Comment (segdeha) [inline src/pages/List.js]: Before merging, please remove commented out code. Thanks!

File: src/pages/List.js (hunk 3 above)
Comment (segdeha) [reply src/pages/List.js]: Nice!

File: src/pages/List.js [hunk 8]
```diff
@@ -1,19 +1,20 @@
...
 
 const List = () => {
+  // TODO: Get the token saved in local storage and use it to filter the list
```
Comment (segdeha) [inline src/pages/List.js]: No longer need this `TODO`, correct?

File: src/pages/AddItem.js (hunk 5 above)
Comment (prophen) [reply src/pages/AddItem.js]: I think I got them all https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/49/commits/1039ba7f558d56e10488afb203aeecf8357f302a

**Comment by prophen** [comment]
@ajiles91 Do you want to give the PR a final look and then merge?

**Pull request merged**